pkg/lib/authenticationflow/declarative/intent_signup_flow_ensure_contraints_fulfilled.go

```diff
@@ -0,0 +1,121 @@
+package declarative
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/authgear/authgear-server/pkg/api/model"
+	"github.com/authgear/authgear-server/pkg/lib/authenticationflow"
+	authflow "github.com/authgear/authgear-server/pkg/lib/authenticationflow"
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/iawaknahc/jsonschema/pkg/jsonpointer"
+)
+
+func init() {
+	authflow.RegisterIntent(&IntentSignupFlowEnsureContraintsFulfilled{})
+}
+
+type IntentSignupFlowEnsureContraintsFulfilled struct {
+	UserID        string                                   `json:"user_id"`
+	FlowObject    *config.AuthenticationFlowSignupFlowStep `json:"flow_object"`
+	FlowReference authenticationflow.FlowReference         `json:"flow_reference,omitempty"`
+	JSONPointer   jsonpointer.T                            `json:"json_pointer,omitempty"`
+}
+
+type IntentSignupFlowEnsureContraintsFulfilledOptions struct {
+	UserID        string                           `json:"user_id"`
+	FlowReference authenticationflow.FlowReference `json:"flow_reference,omitempty"`
+}
+
+func NewIntentSignupFlowEnsureContraintsFulfilled(ctx context.Context, deps *authenticationflow.Dependencies, flows authenticationflow.Flows, opts *IntentSignupFlowEnsureContraintsFulfilledOptions) (*IntentSignupFlowEnsureContraintsFulfilled, error) {
+	var oneOfs []*config.AuthenticationFlowSignupFlowOneOf
+	recoveryCodeStep := &config.AuthenticationFlowSignupFlowStep{
+		Type: config.AuthenticationFlowSignupFlowStepTypeViewRecoveryCode,
+	}
+
+	addOneOf := func(am config.AuthenticationFlowAuthentication, bpGetter func(*config.AppConfig) (*config.AuthenticationFlowBotProtection, bool)) {
+
+		oneOf := &config.AuthenticationFlowSignupFlowOneOf{
+			Authentication: am,
+		}
+
+		if bpGetter != nil {
+			if bp, ok := bpGetter(deps.Config); ok {
+				oneOf.BotProtection = bp
+			}
+		}
+
+		oneOf.Steps = append(oneOf.Steps, recoveryCodeStep)
+		oneOfs = append(oneOfs, oneOf)
+	}
+
+	for _, authenticatorType := range *deps.Config.Authentication.SecondaryAuthenticators {
+		switch authenticatorType {
+		case model.AuthenticatorTypePassword:
+			addOneOf(config.AuthenticationFlowAuthenticationSecondaryPassword, nil)
+		case model.AuthenticatorTypeOOBEmail:
+			addOneOf(config.AuthenticationFlowAuthenticationSecondaryOOBOTPEmail, getBotProtectionRequirementsOOBOTPEmail)
+		case model.AuthenticatorTypeOOBSMS:
+			addOneOf(config.AuthenticationFlowAuthenticationSecondaryOOBOTPSMS, getBotProtectionRequirementsOOBOTPSMS)
+		case model.AuthenticatorTypeTOTP:
+			addOneOf(config.AuthenticationFlowAuthenticationSecondaryTOTP, nil)
+		case model.AuthenticatorTypePasskey:
+			// FIXME(tung): We don't have a step to force user create passkey at the moment
```

Comment:
I found that we don't have a "Create Passkey" option in the create authenticator step at the moment, so if amr contraints includes `x_passkey`, we still cannot ensure user created passkey in signup flow.